File: 01_ProgrammingBasics/_06_NestedLoopsMoreExercises/_09_SumOfTwoNumbers.java
package _01_ProgrammingBasics._06_NestedLoopsMoreExercises;

import java.util.Scanner;

public class _09_SumOfTwoNumbers {
    public static void main(String[] args) {
        Scanner scan = new Scanner(System.in);

        // Read input values
        int startOfInterval = Integer.parseInt(scan.nextLine());
        int endOfInterval = Integer.parseInt(scan.nextLine());
        int magicNumber = Integer.parseInt(scan.nextLine());

        // Add variables
        int combinations = 0;
        boolean magicNumberFound = false;

        // Find the combinations if there are any
        forLoopOuter:
        for (int i = startOfInterval; i <= endOfInterval; i++) {
            for (int j = startOfInterval; j <= endOfInterval; j++) {
                combinations++;
                int sum = i + j;

                if (magicNumber == sum) {
                    magicNumberFound = true;
                    System.out.printf("Combination N:%d (%d + %d = %d)", combinations, i, j, sum);
                    break forLoopOuter;
                }
            }
        }

        // Print a message if there are no combinations
        if (!magicNumberFound) {
            System.out.printf("%d combinations - neither equals %d", combinations, magicNumber);
        }
        
        scan.close();
    }
}

File: 23_C#Basics/_06_NestedLoopsMoreExercises/_09_SumOfTwoNumbers/_09_SumOfTwoNumbers.cs
using System;

public class _09_SumOfTwoNumbers
{
    public static void Main(string[] args)
    {
        // Read input values
        int startOfInterval = int.Parse(Console.ReadLine());
        int endOfInterval = int.Parse(Console.ReadLine());
        int magicNumber = int.Parse(Console.ReadLine());

        // Add variables
        int combinations = 0;
        bool magicNumberFound = false;

        // Find the combination, if any
        for (int i = startOfInterval; i <= endOfInterval && !magicNumberFound; i++)
        {
            for (int j = startOfInterval; j <= endOfInterval; j++)
            {
                combinations++;
                int sum = i + j;

                if (sum == magicNumber)
                {
                    magicNumberFound = true;
                    Console.WriteLine($"Combination N:{combinations} ({i} + {j} = {sum})");
                    break;  // breaks out of the inner loop
                }
            }
        }

        // Print a message if the magic number wasn't found
        if (!magicNumberFound)
        {
            Console.WriteLine($"{combinations} combinations - neither equals {magicNumber}");
        }
    }
}
